src/main/java/seedu/address/model/Model.java

```diff
@@ -44,6 +45,17 @@ public interface Model {
      */
     void setAddressBookFilePath(Path addressBookFilePath);
 
+    /**
+     * Returns the module the user is teaching.
+     *
+     */
+    Mod getTeaching();
+
+    /**
+     * Sets the module the user is teaching.
+     */
+    void setTeaching(Mod module);
+
```

Comment:
I don't think these methods should be part of the Model interface, as they might not be implemented in other models if we choose to extend it.

Reply:
It is necessary for the `Logic` class to fetch the default course. So, it is still there.